### PR TITLE
Correcting links

### PR DESCRIPTION
**Pull request by KathrynN**
Adjusting documentation links.  Currently they self-reference.
(Please check syntax and correct link target!)

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@KathrynN Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-security-oauth&pullRequestId=1813)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-security-oauth&pullRequestId=1813) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by KathrynN** [comment]
This should be an "obvious fix" as per the directions on the page.  If not, please let me know and I can sign the CLA :)

File: spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/expression/OAuth2SecurityExpressionMethods.java [hunk 1]
```diff
@@ -80,7 +80,7 @@ public boolean throwOnError(boolean decision) {
 
 	/**
 	 * Check if the OAuth2 client (not the user) has the role specified. To check the user's roles see
-	 * {@link #clientHasRole(String)}.
+	 * {@link ExpressionUrlAuthorizationConfigurer#hasRole(String)}.
```
Comment (jgrandja) [inline spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/expression/OAuth2SecurityExpressionMethods.java:83]: This should reference `#clientHasAnyRole(String...)`

File: spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/expression/OAuth2SecurityExpressionMethods.java [hunk 2]
```diff
@@ -91,7 +91,7 @@ public boolean clientHasRole(String role) {
 
 	/**
 	 * Check if the OAuth2 client (not the user) has one of the roles specified. To check the user's roles see
-	 * {@link #clientHasAnyRole(String...)}.
+	 * {@link ExpressionUrlAuthorizationConfigurer#hasAnyRole(String...)}.
```
Comment (jgrandja) [inline spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/expression/OAuth2SecurityExpressionMethods.java:94]: This should be `OAuth2ExpressionUtils#clientHasAnyRole(Authentication, String...)`

**Comment by KathrynN** [comment]
@jgrandja sorry for the mistakes and thank you for the review!

**Comment by jgrandja** [comment]
Thanks for the updates @KathrynN. This is now in master.

**Pull request closed without merging**